### PR TITLE
Inject version and job-version into metadata.json

### LINKED ISSUE #1112: Version not showing up on "ci-" builds
On the new "ci-" builds, e.g. [ci-kubernetes-e2e-gce-release-1.5](https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-release-1.5), the "Version" in each of the "ci-" run is missing, for example see [ci-kubernetes-e2e-gce-release-1.5/62/](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-release-1.5/62/). So anago is having trouble finding a build for me to cut a beta.

Same thing is true for other test suites like [ci-kubernetes-e2e-gci-gce-release-1.5](https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-release-1.5)

CC @fejta @ixdy @david-mcmahon @kubernetes/test-infra-admins

### PR DESCRIPTION
**Pull request by fejta**
Testing new version of image at: https://github.com/kubernetes/kubernetes/pull/33463#issuecomment-261126547

Fixes #1112

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1115)

### CODE REVIEW

**Comment by fejta** [comment]
@saad-ali @ixdy @david-mcmahon please lgtm and/or submit at your leisure.

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/33463/pull-kubernetes-e2e-gce/5606

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/33463/pull-kubernetes-e2e-gce/5606/artifacts/metadata.json

which gets stuffed into https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/33463/pull-kubernetes-e2e-gce/5606/finished.json

File: jenkins/e2e-image/e2e-runner.sh [hunk 1]
```diff
@@ -411,7 +411,7 @@ if [[ -n "${BOOTSTRAP_MIGRATION:-}" ]]; then
   # until that time. This environment variable will do that.
   source "$(dirname "${0}")/upload-to-gcs.sh"
   version=$(find_version)  # required by print_started
-  print_started | jq '.metadata?' > "${ARTIFACTS}/metadata.json"
+  print_started | jq '.metadata? + {version, "job-version"}' > "${ARTIFACTS}/metadata.json"
```
Comment (rmmh) [inline jenkins/e2e-image/e2e-runner.sh]: please comment on what this line does, this is spooky

Comment (ixdy) [inline jenkins/e2e-image/e2e-runner.sh]: I agree that this is getting really hard to follow, and it seems pretty fragile.

a comment would be good.

it'd also be nice at some point to have some canonical understanding of what's expected in `started.json`, `metadata.json`, and `finished.json`. it seems to be loosely specified in many different places, which clearly do not all line up.

Comment (fejta) [inline jenkins/e2e-image/e2e-runner.sh]:
```
null + {"foo": 1} == {"foo": 1}
{"bar": 2} + {"foo": 1} == {"bar": 2, "foo": 1}
"string" + {"foo": 1} == 💣 
```

Comment (ixdy) [inline jenkins/e2e-image/e2e-runner.sh]: will this line blow up if we stop including `version` or `job-version` in the output of `print_started`?

Comment (fejta) [inline jenkins/e2e-image/e2e-runner.sh]: Actually I think it will do "version": null
